lib/src/main/java/com/diffplug/spotless/generic/PipeStepPair.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.generic;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.diffplug.spotless.FormatterStep;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class PipeStepPair {
+	/** The two steps will be named `<name>In` and `<name>Out`. */
+	public static Builder named(String name) {
+		return new Builder(name);
+	}
+
+	public static String defaultToggleName() {
+		return "toggle";
+	}
```

Comment:
As far as I can tell, this method is only being used internally in the following fashion:

```java
var pipe = PipeStepPair.named(PipeStepPair.defaultToggleName())...
```

May I suggest replacing it with a method like `PipeStepPair.withDefaultToggleName()` so that all usages can be simplified as follows?

```java
var pipe = PipeStepPair.withDefaultToggleName()...
```

Reply:
I tried that refactor at one point.  The problem is that we still need to expose the step name as a constant, because we remove existing steps by their name.

The broader point is that the pipe step can be used for a lot more than just the toggle:off/on thing that we're using it for now.  It's tempting to overspecialize the API for the one usecase we have currently, but I think we would regret it a little when we implement features like #412.